### PR TITLE
Add bigartm to the docker image

### PR DESCRIPTION
**Pull request by m09**
Building time is long now :crying_cat_face:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,38 +1,64 @@
...
+  libboost-all-dev \
+  make \
+  python3-dev \
+  python3-pip \
+  && ln -s /usr/bin/python3 /usr/local/bin/python \
+  && pip3 install numpy scipy pandas protobuf tqdm wheel \
+  && git clone --branch stable \
+  https://github.com/bigartm/bigartm.git /opt/bigartm \
+  && cd /opt/bigartm \
+  && git reset --hard 3c0cbc4ac08cae37f00754815c105fde930ff379 \
```
Comment (r0mainK) [inline Dockerfile]: Did you check this is needed ? From [here](https://github.com/bigartm/bigartm/commits/stable) it seems it changes nothing, as the last commit reverted the one before, so the module should be in the same state at the HEAD and at commit `3c0cbc4ac08cae37f00754815c105fde930ff379`

File: Dockerfile [hunk 2]
```diff
@@ -1,38 +1,64 @@
...
+  && mkdir build \
+  && cd build \
+  && cmake -DINSTALL_PYTHON_PACKAGE=ON -DPYTHON=python3 .. \
+  && make \
+  && make install \
+  && rm -rf /usr/share/doc /usr/share/man \
```
Comment (r0mainK) [inline Dockerfile:54]: after the `make install` int the [ARTM Dockerfile](https://github.com/bigartm/bigartm-docker/blob/master/Dockerfile) they do the following, do we not need to ?
```
RUN cd 3rdparty/protobuf-3.0.0/python && python3 setup.py build && python3 setup.py install
RUN cd python && python3 setup.py install
```

**Review by r0mainK — CHANGES_REQUESTED** [review]
Got a couple questions but apart from that looks good :)

File: Dockerfile (hunk 1 above)
Comment (m09) [reply Dockerfile]: It's for reproducibility :)

File: Dockerfile (hunk 2 above)
Comment (m09) [reply Dockerfile:54]: I think this one is outdated, last commit is > 2y ago. Here I followed the latest instructions (+ the `-DINSTALL_PYTHON_PACKAGE=ON` that was mentioned nowhere). Though I tested only that `import artm` works, so please let me know if there is any problem and I will investigate!

File: Dockerfile (hunk 1 above)
Comment (r0mainK) [reply Dockerfile]: ah okay. actually, I just realized that this commit hash is actually for the release v0.10.0,
so you can just do `git clone  --branch v0.10.0 https://github.com/bigartm/bigartm.git /opt/bigartm`

File: Dockerfile (hunk 2 above)
Comment (r0mainK) [reply Dockerfile:54]: ah okay, no problem then if something is up we will see :)

File: Dockerfile (hunk 1 above)
Comment (m09) [reply Dockerfile]: Good point!

File: Dockerfile [hunk 3]
```diff
@@ -1,38 +1,63 @@
...
+  python3-pip \
+  && ln -s /usr/bin/python3 /usr/local/bin/python \
+  && pip3 install numpy scipy pandas protobuf tqdm wheel \
+  && git clone --branch v0.10.0 https://github.com/bigartm/bigartm.git /opt/bigartm \
+  && cd /opt/bigartm \
+  && git reset --hard 3c0cbc4ac08cae37f00754815c105fde930ff379 \
+  && mkdir build \
+  && cd build \
+  && cmake -DINSTALL_PYTHON_PACKAGE=ON -DPYTHON=python3 .. \
+  && make \
```
Comment (vmarkovtsev) [inline Dockerfile]:
```suggestion
  && make -j$(getconf _NPROCESSORS_ONLN) \
```
and your build becomes fast again 😄 `getconf` is from `libc-bin`

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (m09) [reply Dockerfile]: somewhat fast since installing liboost-all-dev takes forever :D thanks for the tip!

Comment (vmarkovtsev) [reply Dockerfile]: You don't need to install it btw. Boost is spread into many separate feature packages, so you should install only those which are required.

**Review by r0mainK — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (m09) [reply Dockerfile]: yeah but they don't specify which ones are required and it will take forever to find out. If you have a trick let me know though!

Comment (r0mainK) [reply Dockerfile]: the list of required libs 
```
  libboost-dev \
  libboost-thread-dev \
  libboost-program-options-dev \
  libboost-date-time-dev \
  libboost-filesystem-dev \
  libboost-iostreams-dev \
  libboost-system-dev \
  libboost-chrono-dev \
  libboost-timer-dev \ 
```

**Comment by m09** [comment]
@vmarkovtsev merging despite the changes requested because they were addressed, let me know if you have some other comments :)

**Pull request merged**